src/main/java/com/okta/tools/helpers/FileHelper.java

```diff
@@ -111,15 +110,34 @@ public static Path getFilePath(Path directoryPath, String fileName) throws IOExc
         return filePath;
     }
 
-    /**
-     * Gets the Path of a specified file, without creating it
-     *
-     * @param directoryPath The {@link Path} of the file's parent directory
-     * @param fileName      The name of the file
-     * @return The Path of the file
-     * @throws IOException
-     */
-    public static Path resolveFilePath(Path directoryPath, String fileName) throws IOException {
-        return Paths.get(directoryPath.toString(), fileName);
+    static void usingPath(Path path, PathRW pathRW) throws IOException {
+        try (Reader reader = getReader(path);
+             Writer writer = getWriter(path)) {
+            pathRW.useFile(reader, writer);
+        }
+    }
+
+    public interface PathRW {
+        void useFile(Reader reader, Writer writer) throws IOException;
+    }
+
+    static <T> T readingPath(Path path, PathR<T> pathR) throws IOException {
+        try (Reader reader = getReader(path)) {
+            return pathR.useFile(reader);
+        }
+    }
+
+    public interface PathR<T> {
+        T useFile(Reader reader) throws IOException;
+    }
+
+    static void writingPath(Path path, PathW pathW) throws IOException {
+        try (Writer writer = getWriter(path)) {
+            pathW.useFile(writer);
+        }
+    }
+
+    public interface PathW {
+        void useFile(Writer writer) throws IOException;
```

Comment:
❓ I'm not sure I love the names of these utility methods or interfaces, but I couldn't come up with anything better. I like the outcome in code tho.